src/ImageSharp/Formats/Tiff/Compression/Decompressors/T6TiffCompression.cs

```diff
@@ -0,0 +1,253 @@
+        /// </summary>
+        /// <param name="allocator">The memory allocator.</param>
+        /// <param name="fillOrder">The logical order of bits within a byte.</param>
+        /// <param name="width">The image width.</param>
+        /// <param name="bitsPerPixel">The number of bits per pixel.</param>
+        /// <param name="photometricInterpretation">The photometric interpretation.</param>
+        public T6TiffCompression(
+            MemoryAllocator allocator,
+            TiffFillOrder fillOrder,
+            int width,
+            int bitsPerPixel,
+            TiffPhotometricInterpretation photometricInterpretation)
+            : base(allocator, width, bitsPerPixel)
+        {
+            this.FillOrder = fillOrder;
+            this.width = width;
+            this.isWhiteZero = photometricInterpretation == TiffPhotometricInterpretation.WhiteIsZero;
+            this.whiteValue = (byte)(this.isWhiteZero ? 0 : 1);
+            this.blackValue = (byte)(this.isWhiteZero ? 1 : 0);
+        }
+
+        /// <summary>
+        /// Gets the logical order of bits within a byte.
+        /// </summary>
+        private TiffFillOrder FillOrder { get; }
+
+        /// <inheritdoc/>
+        protected override void Decompress(BufferedReadStream stream, int byteCount, int stripHeight, Span<byte> buffer)
+        {
+            int height = stripHeight;
+
+            using System.Buffers.IMemoryOwner<byte> scanLineBuffer = this.Allocator.Allocate<byte>(this.width * 2);
+            Span<byte> scanLine = scanLineBuffer.GetSpan().Slice(0, this.width);
+            Span<byte> referenceScanLineSpan = scanLineBuffer.GetSpan().Slice(this.width, this.width);
+
+            using var bitReader = new T6BitReader(stream, this.FillOrder, byteCount, this.Allocator);
+
+            var referenceScanLine = new CcittReferenceScanline(this.isWhiteZero, this.width);
+            uint bitsWritten = 0;
+            for (int y = 0; y < height; y++)
+            {
+                scanLine.Fill(0);
+                Decode2DScanline(bitReader, this.isWhiteZero, referenceScanLine, scanLine);
+
+                bitsWritten = this.WriteScanLine(buffer, scanLine, bitsWritten);
+
+                scanLine.CopyTo(referenceScanLineSpan);
+                referenceScanLine = new CcittReferenceScanline(this.isWhiteZero, referenceScanLineSpan);
+            }
+        }
+
+        private uint WriteScanLine(Span<byte> buffer, Span<byte> scanLine, uint bitsWritten)
+        {
+            byte white = (byte)(this.isWhiteZero ? 0 : 255);
+            for (int i = 0; i < scanLine.Length; i++)
+            {
+                BitWriterUtils.WriteBits(buffer, (int)bitsWritten, 1, scanLine[i] == white ? this.whiteValue : this.blackValue);
+                bitsWritten++;
+            }
+
+            // Write padding bytes, if necessary.
+            uint pad = 8 - (bitsWritten % 8);
+            if (pad != 8)
```

Comment:
```suggestion
            uint remainder = bitsWritten % 8;
            if (remainder != 0)
```
Note: here we have unsigned modulo, so the JIT can and will emit the same bit-hack used above.